api/liboni/oni.h

```diff
@@ -79,63 +79,22 @@ typedef struct oni_frame {
 
 } oni_frame_t;
 
-// Context options
-enum {
-    ONI_CONFIGSTREAMPATH,
-    ONI_READSTREAMPATH,
-    ONI_WRITESTREAMPATH,
-    ONI_SIGNALSTREAMPATH,
-    ONI_DEVICEMAP,
-    ONI_NUMDEVICES,
-    ONI_MAXREADFRAMESIZE,
-    ONI_RUNONRESET,
-    ONI_RUNNING,
-    ONI_RESET,
-    ONI_SYSCLKHZ,
-    ONI_BLOCKREADSIZE
-};
-
-// NB: If you add an error here, make sure to update oni_error_str()
-enum {
-    ONI_ESUCCESS         =  0,  // Success
-    ONI_EPATHINVALID     = -1,  // Invalid stream path, fail on open
-    ONI_EDEVID           = -2,  // Invalid device ID on init or reg op
-    ONI_EDEVIDX          = -3,  // Invalid device index
-    ONI_EWRITESIZE       = -4,  // Data write size is incorrect for designated device
-    ONI_EREADFAILURE     = -5,  // Failure to read from a stream/register
-    ONI_EWRITEFAILURE    = -6,  // Failure to write to a stream/register
-    ONI_ENULLCTX         = -7,  // Attempt to call function w null ctx
-    ONI_ESEEKFAILURE     = -8,  // Failure to seek on stream
-    ONI_EINVALSTATE      = -9,  // Invalid operation for the current context run state
-    ONI_EINVALOPT        = -10, // Invalid context option
-    ONI_EINVALARG        = -11, // Invalid function arguments
-    ONI_ECOBSPACK        = -12, // Invalid COBS packet
-    ONI_ERETRIG          = -13, // Attempt to trigger an already triggered operation
-    ONI_EBUFFERSIZE      = -14, // Supplied buffer is too small
-    ONI_EBADDEVMAP       = -15, // Badly formated device map supplied by firmware
-    ONI_EBADALLOC        = -16, // Bad dynamic memory allocation
-    ONI_ECLOSEFAIL       = -17, // File descriptor close failure, check errno
-    ONI_EREADONLY        = -18, // Attempted write to read only object (register, context option, etc)
-    ONI_EUNIMPL          = -19, // Specified, but unimplemented, feature
-    ONI_EINVALREADSIZE   = -20, // Block read size is smaller than the maximal frame size
-    ONI_ENOREADDEV       = -21, // Frame read attempted when there are no readable devices in the device map
-
-    // NB: Always at bottom
-    ONI_MINERRORNUM      = -22
-};
-
 // Context
 typedef struct oni_ctx_impl *oni_ctx;
 
 // Context manipulation
-ONI_EXPORT oni_ctx oni_create_ctx();
-ONI_EXPORT int oni_init_ctx(oni_ctx ctx);
+ONI_EXPORT oni_ctx oni_create_ctx(const char* drivername);
+ONI_EXPORT int oni_init_ctx(oni_ctx ctx, int device_index);
 ONI_EXPORT int oni_destroy_ctx(oni_ctx ctx);
 
 // Context option getting/setting
 ONI_EXPORT int oni_get_opt(const oni_ctx ctx, int option, void* value, size_t *size);
 ONI_EXPORT int oni_set_opt(oni_ctx ctx, int option, const void* value, size_t size);
 
+// Driver option getting/setting
+ONI_EXPORT int_oni_get_driver_opt(const oni_ctx ctx, int driver_option, void* value, size_t *size);
```

Comment:
get rid of underscore after int

Reply:
Also, you should probably turn on -Werror

Reply:
I don't know how I didn't catch this, I could swear I was sure there weren't any warnings on VS...

I didn't try on Linux, I didn't even adapt the Makefile yet (my Linux machine is a VM without access to the PCIe board, so I wanted to get tested first), I'll fix those warnings.

Reply:
its because c implicity declares an int return type, so you got "lucky"